codegen/src/main/java/software/amazon/awssdk/codegen/docs/PaginationDocs.java

```diff
@@ -103,20 +151,56 @@ private String getSyncCodeSnippets() {
                         .toString();
     }
 
+    private String getAsyncCodeSnippets() {
+        CodeBlock callOperationOnClient = CodeBlock.builder()
+                                                   .addStatement("$T publisher = client.$L(request)",
+                                                                 asyncPaginatedResponseType(),
+                                                                 getPaginatedMethodName())
+                                                   .build();
+
+        return CodeBlock.builder()
+                        .add("\n\n<p>The following are few ways to use the response class:</p>")
+                        .add("1) Using the forEach helper method. This uses @{@link $T} internally",
+                             TypeName.get(SequentialSubscriber.class))
+                        .add(buildCode(CodeBlock.builder()
+                                                .add(callOperationOnClient)
+                                                .add(CodeBlock.builder()
+                                                              .addStatement("CompletableFuture<Void> future = publisher"
```

Comment:
Should the future here be fulfilled with the last response? Would that be useful or confusing?